modules/vmseries/main.tf

```diff
@@ -111,6 +111,8 @@ resource "azurerm_virtual_machine" "this" {
     type         = var.identity_type
     identity_ids = var.identity_ids
   }
+  
+  depends_on = [ azurerm_network_interface_backend_address_pool_association.this ]
```

Comment:
I'm doubtful of any explicit depends_on by now. Honestly it has bitten me many times. I'm worried if it will cause any strange problems down the road or not.

Reply:
Let's operate on facts, not feelings. Our vmseries with loadbalancer modules doesn't teardown properly. Point and document any use case that it might raise other issue or just write an issue to azurerm repo and ask them for repair, but for now we need fix for it.

Reply:
All right, you seem optimistic about it after all. What situations did you test? I mean possibilities like: Destroys? Modifications? Additions (i.e. add new interface with lb_ba_pool when some interfaces already exist)? Were there any errors?

When I have not seen any tests, I have defaulted to veto. Don't get me wrong, if this approach actually solves the problem and does not add a worse problem, I will decidedly agree and change my vote.